### PR TITLE
[migration phase 1] Implement CheckVolumeBinding as a filter plugin

### LINKED ISSUE #83635: [migration phase 1] VolumeBindingChecker as filter plugin
**What would you like to be added**:

Add a filter plugin that calls into VolumeBindingChecker predicate, example #83460

**Why is this needed**:

Part of  #83554

/sig scheduling
/help
/priority important-soon

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Implement CheckVolumeBinding as a filter plugin. Later, we can convert all volume binding code into this plugin.

**Which issue(s) this PR fixes**:

Fixes #83635

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by cofyc** [comment]
cc @msau42

**Comment by cofyc** [comment]
/priority important-soon
same as other migration PRs

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+func (pl *VolumeBinding) Filter(cs *framework.CycleState, pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *framework.Status {
+	_, reasons, err := pl.predicate(pod, nil, nodeInfo)
+	return migration.PredicateResultToFrameworkStatus(reasons, err)
+}
+
+// NewFromVolumeBinder initializes a new plugin with volume binder and returns it.
+func NewFromVolumeBinder(volumeBinder *volumebinder.VolumeBinder) (framework.Plugin, error) {
+	return &VolumeBinding{
+		predicate: predicates.NewVolumeBindingPredicate(volumeBinder),
+	}, nil
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go]: nit, remove the error as a returned value since it is always ```nil```

**Review by ahg-g — COMMENTED** [review]
Thanks, one nit. note that you may need to rebase after #83662 gets merged.

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go (hunk 1 above)
Comment (cofyc) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go]: fixed and also rebased on latest master

**Pull request merged**